### PR TITLE
[Test] Add test that revokes OAuth access from Dashboard

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
This test automate revoke OAuth access from User Preferences page.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-4841

### PR Checklist

<code>
        ▼ UserPreferences.openUserPreferencesPage
            ‣ DriverHelper.waitAndClick - By(xpath, //header//button/span[text()!=""]//parent::button)
            ‣ DriverHelper.waitVisibility - By(xpath, //header//button/span[text()!=""]//parent::button)
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitAndClick - By(xpath, //button[text()="User Preferences"])
            ‣ DriverHelper.waitVisibility - By(xpath, //button[text()="User Preferences"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitVisibility - By(xpath, //h1[text()="User Preferences"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
          ▼ UserPreferences.openGitServicesTab
            ‣ DriverHelper.waitAndClick - By(xpath, //button[text()="Git Services"])
            ‣ DriverHelper.waitVisibility - By(xpath, //button[text()="Git Services"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
          ▼ UserPreferences.revokeGitService
          ▼ UserPreferences.selectListItem - of the 'GitHub' list item
            ‣ DriverHelper.waitAndClick - By(xpath, //tr[td[text()='GitHub']]//input)
            ‣ DriverHelper.waitVisibility - By(xpath, //tr[td[text()='GitHub']]//input)
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitAndClick - By(xpath, //button[text()="Revoke"])
            ‣ DriverHelper.waitVisibility - By(xpath, //button[text()="Revoke"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitVisibility - By(xpath, //span[text()="Revoke Git Services"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitAndClick - By(xpath, //input[@data-testid="warning-info-checkbox"])
            ‣ DriverHelper.waitVisibility - By(xpath, //input[@data-testid="warning-info-checkbox"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitAndClick - By(xpath, //button[@data-testid="revoke-button" and not(@disabled)])
            ‣ DriverHelper.waitVisibility - By(xpath, //button[@data-testid="revoke-button" and not(@disabled)])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitDisappearance - By(xpath, //tr[td[text()='GitHub']]//input)
            ‣ DriverHelper.waitDisappearanceBoolean - By(xpath, //tr[td[text()='GitHub']]//input)
            ‣ DriverHelper.isVisible - By(xpath, //tr[td[text()='GitHub']]//input)
            ‣ DriverHelper.wait - (1000 milliseconds)
            ‣ DriverHelper.isVisible - By(xpath, //tr[td[text()='GitHub']]//input)
    ✔ Revoke OAuth test
</code>

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
@SkorikSergey: do you have successful RevokeOauth.spec.ts test run to review?

File: tests/e2e/specs/miscellaneous/RevokeOauth.spec.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { e2eContainer } from '../../configs/inversify.config';
+import { CLASSES } from '../../configs/inversify.types';
+import { LoginTests } from '../../tests-library/LoginTests';
+import { BASE_TEST_CONSTANTS } from '../../constants/BASE_TEST_CONSTANTS';
+import { UserPreferences } from '../../pageobjects/dashboard/UserPreferences';
+
+suite(`"Check User Preferences page" test ${BASE_TEST_CONSTANTS.TEST_ENVIRONMENT}`, function (): void {
```
Comment (dmytro-ndp) [inline tests/e2e/specs/miscellaneous/RevokeOauth.spec.ts]: Could you please say what "${BASE_TEST_CONSTANTS.TEST_ENVIRONMENT}" is for?

It's expected to be one of the git provider, isn't it?

File: tests/e2e/specs/miscellaneous/RevokeOauth.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	suiteSetup('Login', async function (): Promise<void> {
+		await loginTests.loginIntoChe();
+	});
+
+	test('Check user preferences page', async function (): Promise<void> {
+		await userPreferences.openUserPreferencesPage();
+		await userPreferences.checkTabsAvailability();
+
+		await userPreferences.openGitServicesTab();
+		await userPreferences.revokeGitService('GitHub');
```
Comment (dmytro-ndp) [inline tests/e2e/specs/miscellaneous/RevokeOauth.spec.ts]: Let's make the test unified, to be able to accept other git providers like gitlab, etc.

**Comment by SkorikSergey** [comment]
> @SkorikSergey: do you have successful RevokeOauth.spec.ts test run to review?

Yes, checked with GitHub service.

File: tests/e2e/specs/miscellaneous/RevokeOauth.spec.ts (hunk 1 above)
Comment (SkorikSergey) [reply tests/e2e/specs/miscellaneous/RevokeOauth.spec.ts]: I don't know :). BASE_TEST_CONSTANTS.TEST_ENVIRONMENT is not set in constants and I don't see any using it in crw-jenkins jobs.

File: tests/e2e/pageobjects/dashboard/UserPreferences.ts [hunk 3]
```diff
@@ -14,14 +14,25 @@ import { By } from 'selenium-webdriver';
 import { DriverHelper } from '../../utils/DriverHelper';
 import { Logger } from '../../utils/Logger';
 
+enum GitServices {
+	GITHUB = 'github',
```
Comment (dmytro-ndp) [inline tests/e2e/pageobjects/dashboard/UserPreferences.ts]: IMHO we can avoid duplication if use FACTORY_TEST_CONSTANTS > GitProviderType https://github.com/eclipse/che/blob/07ed023e6d80e46c9a1759d7ff25a77d4e9ab535/tests/e2e/constants/FACTORY_TEST_CONSTANTS.ts#L12-L18

File: tests/e2e/pageobjects/dashboard/UserPreferences.ts [hunk 4]
```diff
@@ -86,4 +119,21 @@ export class UserPreferences {
 		await this.driverHelper.waitAndClick(UserPreferences.SSH_KEY_TAB);
 		await this.driverHelper.waitVisibility(UserPreferences.ADD_NEW_SSH_KEY_BUTTON);
 	}
+
+	getServiceConfig(service: string): string {
+		const gitService: { [key: string]: string } = {
+			[GitServices.GITHUB]: 'GitHub',
```
Comment (dmytro-ndp) [inline tests/e2e/pageobjects/dashboard/UserPreferences.ts]: See comment in https://github.com/eclipse/che/pull/22762/files#r1452432608

**Review by dmytro-ndp — APPROVED** [review]
Looks commonly good to merge, with a small room for improvement commented earlier.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**